core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+   * fractions.
+   * * Fractional sweep of the arc is the smaller of the cyclic sweeps.
+   */
+  protected updateMaxPerpendicular(childApproximation: Arc3d | LineSegment3d, f0: number, f1: number): void {
+    const childPerp = ErrorProcessor.computePrimitiveErrorXY(
+      this.ellipticalArc.clonePartialCurve(f0, f1), childApproximation,
+    );
+    if (childPerp && (!this.maxPerpendicular || this.maxPerpendicular.detailA.a < childPerp.detailA.a))
+      this.maxPerpendicular = childPerp;
+  };
+
+};
+
+/**
+ * Processor for computing the error of a sample-based arc chain approximation.
+ * @internal
+ */
+class ArcChainErrorProcessor extends ChainErrorProcessor {
+  public constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+  }
+  public override announceArc(arc: Arc3d, f0: number, f1: number): void {
+    this.updateMaxPerpendicular(arc, f0, f1);
+  }
+}
+
+/**
+ * Processor for computing the error of a sample-based linestring approximation.
+ * @internal
+ */
+class LineStringErrorProcessor extends ChainErrorProcessor {
+  public constructor(ellipticalArc: Arc3d) {
+    super(ellipticalArc);
+  }
+  public override announceChord(pt0: Point3d, pt1: Point3d, f0: number, f1: number): void {
+    this.updateMaxPerpendicular(LineSegment3d.create(pt0, pt1), f0, f1);
+  }
+}
+
+/**
+ * Intermediate processor class for computing samples for a subdivision-based approximation.
+ * @internal
+ */
+class SubdivisionErrorProcessor extends ErrorProcessor {
+  private _refinement: OrderedSet<number>;
+  private _maxError: number;
+  private _originalRefinementCount: number;
+  protected constructor(ellipticalArc: Arc3d, maxError: number) {
+    super(ellipticalArc);
+    this._refinement = new OrderedSet<number>(compareFractions);
+    this._maxError = maxError > 0 ? maxError : EllipticalArcApproximationOptions.defaultMaxError;
+    this._originalRefinementCount = 0;
+  }
+  public override announceQuadrantBegin(q: QuadrantFractions): boolean {
+    this._refinement.clear();
+    for (const f of q.fractions)
+      this._refinement.add(f);
+    this._originalRefinementCount = this._refinement.size;
+    return true;
+  }
+  public updateFractionSet(childApproximation: Arc3d | LineSegment3d, f0: number, f1: number): void {
+    const childPerp = ErrorProcessor.computePrimitiveErrorXY(
+      this.ellipticalArc.clonePartialCurve(f0, f1), childApproximation,
+    );
+    if (childPerp && (childPerp.detailA.a > this._maxError))
+      this._refinement.add(Geometry.interpolate(f0, 0.5, f1));
+  }
+  public override announceQuadrantEnd(q: QuadrantFractions, _wasReversed: boolean): void {
+    q.fractions = [...this._refinement];
+  }
+  /** Whether the processor refined the sample set to decrease the approximation error. */
+  public get isRefined(): boolean {
+    return this._originalRefinementCount < this._refinement.size;
+  }
+  public getRefinedAngles(result?: number[]): number[] {
```

Comment:
similarly what the `result` array?

Reply:
I'll document. But its usage further below clarifies.